tests/test_lime_survey.py

```diff
@@ -208,6 +209,103 @@ def test_lime_system_info_dtypes(self):
             True,
         )
 
+class TestLimeSurveyGetResponse(BaseTestLimeSurvey2021Case):
+    """Test LimeSurvey get response"""
+
+    def setUp(self) -> None:
+        """Read responses before each test."""
+        self.survey.read_responses(responses_file=self.responses_file)
+
+    def test_get_response(self):
+        """Test basic get response functionality for valid and invalid input"""
+        try:
+            questions = [
+                'A1', 'A3', 'A4', 'A5', 'A10',
+                'A14', 'B6', 'D2', 'E5a', 'E9'
+            ]
+            for question in questions:
+                self.survey.get_responses(question, labels=False)
+                self.survey.get_responses(question, labels=True)
+        except:
+            self.fail('No assertion expected for valid input.')
+
+        # assume exception raised after invalid question name
+        self.assertRaises(ValueError, self.survey.get_responses, 'X2', labels=False)
+        self.assertRaises(ValueError, self.survey.get_responses, 'X2', labels=True)
+
+        # assume exception is thrown after A2 missing in responses
+        self.assertRaises(KeyError, self.survey.get_responses, 'A2', labels=False)
+        self.assertRaises(KeyError, self.survey.get_responses, 'A2', labels=True)
+
+    def test_get_response_single_choice(self):
+        """Test get response for single choice question type"""
+        expected_response = [
+            'A1', 'A1', 'A3', 'A2', 'nan',
+            'A1', 'A1', 'A1', 'A1', 'A1'
+        ]
+        response = self.survey.get_responses(self.single_choice_column, labels=False)
+        np.testing.assert_array_equal(
+            expected_response,
+            response.values.flatten().astype(str)[:10]
```

Comment:
You can "cut" the data frame in `setUpClass` (I mean like `s.responses = s.responses.iloc[:10,:]`, so you do not have to do it in all tests.

Reply:
As far as I know, setUpClass is called once before all tests of that class. setUp once before each test. So, if I check different responses this will not work.